media/src/Settings.svelte

```diff
@@ -115,72 +122,149 @@
         class="btn btn-sm btn-light box-3"
         style="margin-left: auto;"
         on:click={() => {
-        showSettings = false;
-        }}>
+            showSettings = false;
+        }}
+    >
         <i class="bi bi-x-lg"></i>
     </button>
 
-    <span class="box-1 text-right">scale</span>
-    <span class="box-23"
-          ><input
-               type="range"
-               name="scale"
-               id="scale"
-               min="-2"
-               max="3"
-               step=".02"
-               bind:value={scale}
-               on:change={rescale}
-               /><span class="output">{scala}</span></span>
+    <label class="form-label" for="scale">Scale</label>
+    <span class="form-range">
+        <input
+            type="range"
+            name="scale"
+            id="scale"
+            min="-2"
+            max="3"
+            step=".02"
+            bind:value={scale}
+            on:change={rescale}
+        >
+        <span class="output box-23 text-right">{scala}</span>
+    </span>
 
-    <span class="box-1 text-right">grid</span>
-    <label class="switch box-2">
+    <div class="form-check form-switch">
+        <label class="form-check-label box-1" for="gridVisible">Grid</label>
         <input
+            class="form-check-input box-1"
             type="checkbox"
             name="gridVisible"
             id="gridVisible"
+            role="switch"
+            aria-checked="true"
             bind:checked={gridMeshes.visible}
             on:change={render}
-            />
-        <span class="slider round" />
-    </label>
+        />
+    </div>
 
-    <span class="box-1 text-right">ortho/proj</span>
-    <label class="switch box-2">
+    <div class="form-check form-switch">
+        <label class="form-check-label box-2" for="orthoCamera">Linear Perspective</label>
         <input
+            class="form-check-input box-2"
             type="checkbox"
+            role="switch"
+            aria-checked="false"
             name="orthoCamera"
             id="orthoCamera"
             bind:checked={orthoCamera}
             on:change={render}
-            />
-        <span class="slider round" />
-    </label>
+        />
+    </div>
+</div>
+
+<div
+    class="settings-box"
+    class:grid={showUpload}
+    hidden={!showUpload}
+    id="upload-box"
+>
+    <button
+        type="button"
+        class="btn btn-sm btn-light box-3"
+        style="margin-left: auto;"
+        on:click={() => {
+            showUpload = false;
+        }}
+    >
+        <i class="bi bi-x-lg"></i>
+    </button>
+    <label for="upload">Upload a scene</label>
+    <input id="upload" type="file" accept="application/json"
+        on:change={(e) => {
```

Comment:
This is a pretty complex anonymous function defined inline in the template here - I think it should be re-organized out into a proper function, like:
```
<input on:change={handleSceneImport} />
```

Reply:
Minor tweak to your suggestion: I had to pass the event target to the function so that it know where to find the file.